ethpm_cli/manager.py

```diff
@@ -0,0 +1,87 @@
+from ethpm.typing import Manifest
+from eth_utils import to_text
+from ethpm.backends.ipfs import BaseIPFSBackend
+from ethpm.utils.ipfs import is_ipfs_uri
+import json
+from pathlib import Path
+from typing import Dict
+
+from eth_utils.toolz import assoc
+from ethpm.typing import URI
+
+from ethpm_cli.exceptions import InstallError
+from ethpm_cli.package import Package
+from ethpm_cli._utils.ipfs import get_ipfs_backend
+
+
+class Manager:
+    def __init__(self, target_dir: Path = None, ipfs: bool = None) -> None:
+        self.ethpm_dir = process_target_dir(target_dir)
+        self.ethpm_lock = self.ethpm_dir / "ethpm.lock"
+        self.ipfs_backend = get_ipfs_backend(ipfs)
+
+    def install(self, target_uri: URI, alias: str = None) -> None:
+        target_pkg = Package(target_uri, alias, self.ipfs_backend)
+        pkg_path = self.ethpm_dir / target_pkg.alias
+        if pkg_path.is_dir():
+            raise InstallError(f"{target_pkg.alias} is already installed.")
+        pkg_path.mkdir()
```

Comment:
Assuming that `alias` is a valid package name (which is provided from cli arg validation) - Is it not safe to assume that `pkg_path` is always strictly under `self.ethpm_dir` without doing additional validation? Or am I missing something?

Reply:
It's more that the validation of `alias` is implicit and since this would be a major-ish security vulnerability I'd like to have this explicitely validated independently.